packages/sparrowql/src/build.ts

```diff
@@ -308,21 +319,32 @@ export function prepare({
       return;
     }
 
-    const available = Object.keys(query).filter(key =>
-      isOperator(key)
-        ? (query[key] as RequiredQuery).required.every(key =>
+    const isMongoQuery = (
+      key: string,
+      query: MongoQuery | PerformQuery,
+    ): query is MongoQuery => isOperator(key);
+    const isPerformQuery = (
+      key: string,
+      query: MongoQuery | PerformQuery,
+    ): query is PerformQuery => isOperator(key);
+
+    const available = Object.keys(query).filter(key => {
+      const singleQuery = query[key];
+      return isMongoQuery(key, singleQuery)
+        ? singleQuery.required.every(key =>
             joined.includes(getNameCollection(key)),
           )
-        : joined.includes(getNameCollection(key)),
-    );
+        : joined.includes(getNameCollection(key));
```

Comment:
How about this?
```ts
const keys = isMongoQuery(key, singleQuery) ? singleQuery.required : [key];
return keys.every(key => joined.includes(getNameCollection(key)));
```

Reply:
Unfortunately it gets wrapped into:
```ts
      const keys = isMongoQuery(key, singleQuery)
        ? singleQuery.required
        : [key];
      return keys.every(key => joined.includes(getNameCollection(key)));
      ```

But it still looks much better to me than the previous version